apps/web/src/pages/integrations/components/multi-provider/SelectPrimaryIntegrationModal.tsx

```diff
@@ -0,0 +1,261 @@
+  }, [integrations, environments, channelType, environmentId]);
+
+  const initialSelectedIndex = useMemo(() => {
+    const primary = findPrimaryIntegration(integrationsByEnvAndChannel ?? []);
+
+    return integrationsByEnvAndChannel?.indexOf(primary);
+  }, [integrationsByEnvAndChannel]);
+
+  const isLoading = areEnvironmentsLoading || areIntegrationsLoading;
+  const isInitialProviderSelected = !selectedRowId || selectedRowId === `${initialSelectedIndex}`;
+  const makePrimaryButtonDisabled = !selectedIntegrationId || isLoading || isInitialProviderSelected;
+  const channelName = CHANNEL_TYPE_TO_STRING[channelType ?? ''];
+
+  const onRowSelectCallback = (row: Row<ITableIntegration>) => {
+    setSelectedState({
+      selectedIntegrationId: row.original.integrationId,
+      isActive: row.original.active,
+      selectedRowId: row.id,
+    });
+  };
+
+  return (
+    <Modal
+      opened={isOpened}
+      withinPortal
+      overlayColor={theme.colorScheme === 'dark' ? colors.BGDark : colors.BGLight}
+      overlayOpacity={0.7}
+      styles={{
+        root: {
+          backdropFilter: 'blur(10px)',
+        },
+        modal: {
+          backgroundColor: theme.colorScheme === 'dark' ? colors.B15 : colors.white,
+          width: 800,
+          '@media (max-width: 1366px)': {
+            maxHeight: '650px',
+          },
+        },
+        inner: {
+          alignItems: 'center',
+        },
+        header: {
+          marginBottom: '16px',
+        },
+      }}
+      title={<Title size={2}>Select primary provider</Title>}
+      shadow={theme.colorScheme === 'dark' ? shadows.dark : shadows.medium}
+      radius="md"
+      size="lg"
+      onClose={onClose}
+    >
+      <ModalBodyHolder data-test-id="select-primary-integration-modal">
+        <Description>
+          Please select the primary provider instance within the email channel in the development environment.
+        </Description>
+        <MetaHolder>
+          <IntegrationEnvironmentPill
+            name={environmentName}
+            testId="integration-environment-pill"
+            isLoading={isLoading}
+          />
+          <IntegrationChannel
+            type={channelType}
+            name={channelName}
+            sameColor
+            isLoading={isLoading}
+            testId="integration-channel-cell"
+          />
+        </MetaHolder>
+        <When truthy={isLoading}>
+          <Table loading loadingItems={5} data-test-id="integrations-list-table" columns={columns} withSelection />
+        </When>
+        <When truthy={!isLoading}>
+          <Table
+            data-test-id="integrations-list-table"
+            columns={columns}
+            data={integrationsByEnvAndChannel}
+            withSelection
+            onRowSelect={onRowSelectCallback}
+            initialSelectedRows={initialSelectedIndex > -1 ? { [initialSelectedIndex]: true } : undefined}
+          />
+        </When>
```

Comment:
the trick to have a initially selected row... in the beginning, we don't have data for integrations, so we cant calculate the selected row index... if we would set it to -1 in the begging then after the data arrives that index value would be different ex. 1, but the state won't be updated as it's just used for the init... 

that's why we do show one table on loading and another when the data is ready...